examples/multiplayer-lan/rust/src/player.rs

```diff
@@ -0,0 +1,215 @@
+use godot::classes::{
+    Area2D, CharacterBody2D, ICharacterBody2D, MultiplayerSynchronizer, PhysicsBody2D,
+    ProjectSettings,
+};
+use godot::global::move_toward;
+use godot::obj::WithBaseField;
+use godot::prelude::*;
+
+use crate::bullet::Bullet;
+use crate::NetworkId;
+
+const MAX_HEALTH: i32 = 2;
+
+#[derive(GodotClass)]
+#[class(base=CharacterBody2D)]
+pub struct Player {
+    speed: f32,
+    jump_velocity: f32,
+    gravity: f64,
+    #[export]
+    health: i32,
+    #[export]
+    bullet_scene: Gd<PackedScene>,
+    // multiplayer stuff
+    multiplayer_synchronizer: OnReady<Gd<MultiplayerSynchronizer>>,
+    #[export]
+    pub username: GString,
+    #[var]
+    pub network_id: NetworkId,
+    #[export]
+    sync_position: Vector2,
+    #[export]
+    sync_rotation: f32,
+    base: Base<CharacterBody2D>,
+}
+
+#[godot_api]
+impl Player {
+    #[signal]
+    fn death();
+
+    #[rpc(any_peer, call_local)]
+    fn fire(&self) {
+        let mut bullet = self.bullet_scene.instantiate_as::<Bullet>();
+        bullet.set_global_position(
+            self.base()
+                .get_node_as::<Node2D>("GunRotation/BulletSpawn")
+                .get_global_position(),
+        );
+        bullet.set_rotation_degrees(
+            self.base()
+                .get_node_as::<Node2D>("GunRotation")
+                .get_rotation_degrees(),
+        );
+        bullet.bind_mut().set_network_id(self.network_id);
+        self.base()
+            .get_tree()
+            .unwrap()
+            .get_root()
+            .unwrap()
+            .add_child(bullet);
+    }
+
+    /*
+    // only the server/host player can call this function
+    #[rpc(authority, call_local, reliable)]
+    fn respawn(&mut self, position: Vector2, network_id: NetworkId){
+        if self.network_id == network_id {
+            self.health = MAX_HEALTH;
+            self.set_player_position_from_server(position, network_id);
+        }
+    }
+    */
+
+    // only the server/host player can call this function
+    #[rpc(authority, call_local, reliable)]
+    fn set_player_position_from_server(&mut self, position: Vector2, network_id: NetworkId) {
+        if self.network_id == network_id {
+            self.base_mut().set_global_position(position);
+            self.set_sync_position(position);
+        }
+    }
+
+    // Tried to make this an actual game by having a respawn system and health.
+    // TODO: Figure out how to make this work
```

Comment:
TODO in an example?

Reply:
Yeah i still need to implement this, but the syncing errors proved too annoying so I had to figure out a way to deal with that first